src/ray/raylet/worker_pool.cc

```diff
@@ -404,8 +404,13 @@ std::shared_ptr<Worker> WorkerPool::PopWorker(const TaskSpecification &task_spec
 
 bool WorkerPool::DisconnectWorker(const std::shared_ptr<Worker> &worker) {
   auto &state = GetStateForLanguage(worker->GetLanguage());
-  RAY_CHECK(RemoveWorker(state.registered_workers, worker));
+  // Once a worker disconnected, remove the starting workers of the same process.
+  auto it = state.starting_worker_processes.find(static_cast<int>(worker->Pid()));
+  if (it != state.starting_worker_processes.end()) {
+    state.starting_worker_processes.erase(it);
```

Comment:
why should we remove all the workers in this process? Should we instead do the removal for each worker when it disconnects?

Reply:
Could you add some descriptions in the PR on what is the problem and how this fix solves it?